### ISSUE TITLE
NO PRINT IMPLEMENTED: ObjectKeyword

### ISSUE THREAD

**Issue by stereobooster** [joarwilk/flowgen#35]
Seeing this error when trying to convert typings of downshift https://github.com/paypal/downshift/pull/372

Related: https://github.com/joarwilk/flowgen/issues/5

**Comment by orta** [comment]
Hi, cool, are you interested in fixing this or writing a test for it?

**Comment by stereobooster** [comment]
I'm interested in fixing it if/when I have time.

**Comment by stereobooster** [comment]
> TypeScript 2.2 introduced a new type called object. It represents any non-primitive type. The following types are considered to be primitive types in JavaScript:

```ts
// All primitive types
type Primitive =
    | boolean
    | number
    | string
    | symbol
    | null
    | undefined;

// All non-primitive types
type NonPrimitive = object;
```
Does anybody knows what is corresponding type in Flow? Is it `{}`?

**Comment by saadq** [comment]
Flow has `Object`, but I think `{}` works too

**Comment by stereobooster** [comment]
TypeScript has `Object`, `object` and `{}`. I suppose `Object` and `{}` in Flow are pretty close to what TypeScript means, not so sure about `object`.

**Comment by saadq** [comment]
I think Flow's `Object` is equivalent to TypeScript's `object`. And yeah, `object` doesn't exist in Flow.

**Comment by stereobooster** [comment]
> I think Flow's Object is equivalent to TypeScript's object

I don't think so. Definition of [`Object`](https://github.com/facebook/flow/blob/v0.67.0/lib/core.js#L38).

TS `object` any non-primitive value, for example `Date`.

```js
(new Date()).create // undefined
```

**Comment by saadq** [comment]
I think that's definition is only used when you create an object using the constructor like

```js
let obj = new Object()
```

But when you use `Object` as a type annotation, it basically means any non-primitive. I found someone who has a fork of the old Flow docs, and it seems to agree: https://zhenyong.github.io/flowtype/docs/objects.html#the-object-type

The Flow docs update was nice, but I think a lot of valuable info was lost from it like this and callable objects, etc...

**Comment by goodmind** [comment]
Fixed in 9a94bcca48de2d30fc15869329331712d17f43a1
Do you think this is the same as TS `object`?
